rmgpy/reaction.py

```diff
@@ -1123,3 +990,62 @@ def copy(self):
         
         return other
 
+def _isomorphicSpeciesList(list1, list2, checkIdentical=False):
+    """
+    This method compares whether lists of species or molecules are isomorphic
+    or identical. It is used for the 'isIsomorphic' method of Reaction class.
+    It likely can be useful elswehere as well:
+        
+        list1 - list of species/molecule objects of reaction1
+        list2 - list of species/molecule objects of reaction2
+        checkIdentical - if true, uses the 'isIdentical' comparison
+                         if false, uses the 'isIsomorphic' comparison
+                         
+    Returns True if the lists are isomorphic/identical & false otherwise
+    """
+
+    def comparison_method(other1, other2, checkIdentical=checkIdentical):
+        if checkIdentical:
+            return other1.isIdentical(other2)
+        else:
+            return other1.isIsomorphic(other2)
+
+    if len(list1) == len(list2) == 1:
+        if comparison_method(list1[0], list2[0]):
+            return True
+    elif len(list1) == len(list2) == 2:
+        if comparison_method(list1[0], list2[0]) \
+                    and comparison_method(list1[1], list2[1]):
+            return True
+        elif comparison_method(list1[0], list2[1]) \
+                    and comparison_method(list1[1], list2[0]):
+            return True
+    elif len(list1) == len(list2) == 3:
```

Comment:
I like the new approach, but I don't think the time savings would be as significant as you would expect. See the table below for a comparison of the number of function calls. Note that `list.remove` performs an isomorphism comparison with each item in the list until it finds the one it needs to remove. Using `enumerate` along with `del`  instead of `list.remove` might help a little bit.

reactants | case | if/elif method | loop method
---------- | ----- | ---------- | ----------
1 | | 1 comparison call | 1 comparison call <br> 1 list.remove call
2 | best | 2 comparison calls | 2 comparison calls <br> 2 list.remove calls
2 | worst | 4 comparison calls | 3 comparison calls <br> 2 list.remove calls
3 | best | 3 comparison calls | 3 comparison calls <br> 3 list.remove calls
3 | worst | 18 comparison calls | 6 comparison calls <br> 3 list.remove calls